test/Riok.Mapperly.Tests/Mapping/EnumNamingStrategyTest.cs

```diff
@@ -0,0 +1,172 @@
+namespace Riok.Mapperly.Tests.Mapping;
+
+public class EnumNamingStrategyTest
+{
+    [Fact]
+    public void EnumToStringWithPascalCaseNamingStrategy()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapEnum(EnumMappingStrategy.ByName, NamingStrategy = EnumNamingStrategy.PascalCase)] public partial string ToStr(E source);",
+            "public enum E {Abc, BcD, C1D, DEf, EFG, FG1, Gh1, Hi_J}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    global::E.Abc => "Abc",
+                    global::E.BcD => "BcD",
+                    global::E.C1D => "C1D",
+                    global::E.DEf => "DEf",
+                    global::E.EFG => "EFG",
+                    global::E.FG1 => "FG1",
+                    global::E.Gh1 => "Gh1",
+                    global::E.Hi_J => "Hi_J",
+                    _ => source.ToString(),
+                };
+                """
+            );
+    }
+
+    [Fact]
+    public void EnumToStringWithSnakeCaseNamingStrategy()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapEnum(EnumMappingStrategy.ByName, NamingStrategy = EnumNamingStrategy.SnakeCase)] public partial string ToStr(E source);",
+            "public enum E {Abc, BcD, C1D, DEf, EFG, FG1, Gh1, Hi_J}"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveSingleMethodBody(
+                """
+                return source switch
+                {
+                    global::E.Abc => "abc",
+                    global::E.BcD => "bc_d",
+                    global::E.C1D => "c_1_d",
+                    global::E.DEf => "d_ef",
+                    global::E.EFG => "efg",
+                    global::E.FG1 => "fg_1",
+                    global::E.Gh1 => "gh_1",
+                    global::E.Hi_J => "hi_j",
+                    _ => source.ToString(),
```

Comment:
What happens in the default/fallback case 🤔 IMO we would need to transform this too or add to the documentation that the default fallback is not transformed.

Reply:
I think that users might want to use fallback values like "_default" that are only possible with the default strategy. I have added documentation to clarify that the fallback value remains unchanged by the strategies.